src/model.rs

```diff
@@ -61,6 +61,30 @@ impl Model {
         unsafe { std::slice::from_raw_parts((*self.model).q, (self.columns() * self.factors()) as usize) }
     }
 
+    pub fn p_row(&self, row_index: i32) -> Option<&[f32]>{
+        use std::convert::TryFrom;
+        if let Ok(factors) = usize::try_from( self.factors() ){
```

Comment:
What's the reasoning behind using `try_from`?

Reply:
It was to not panic if there were negative factors. But that shouldn't happen, and it probably should panic if it does, so I've removed it.